lib/Targets/AIETargetXAIEV2.cpp

```diff
@@ -37,6 +37,17 @@ using namespace xilinx::AIEX;
 namespace xilinx {
 namespace AIE {
 
+// This string is output at the top of the lowered C code.
+const char xaie_c_file_header[] =
+    "#ifndef MLIR_AIE_QUIET\n"
+    "#define __mlir_aie_print(x) x\n"
+    "#else\n"
+    "#define __mlir_aie_print(x)\n"
+    "#endif\n"
+    "#define __mlir_aie_try(x) do { AieRC ret = (x); if(ret != XAIE_OK) { "
+    "__mlir_aie_print(fprintf(stderr, #x \"failed: %d\\n\", x)); return x; } "
+    "} while(0)\n";
```

Comment:
You could use raw strings (see example of https://en.cppreference.com/w/cpp/language/string_literal, the equivalent of Python `"""` multi-line strings) and format normally the code instead of the current obfuscation.
It took me some time to understand the logic of the code because of the `return` I misinterpreted.
The fact that the `return` can actually be used as a kind of error exit here is super related to the way the runtime is generated on the fly. If it changes some day, it will break the code. I do not say it is bad. At least this could appear in a comment inside the macro for the next person refactoring the runtime architecture. :-)
You would probably want to print the line number, function, file, etc.

Reply:
This is a good suggestion, I will make this change so the code is more readable.

What do you mean by this:
> The fact that the return can actually be used as a kind of error exit here is super related to the way the runtime is generated on the fly.

I think the return can be used as an "exit" because it is inside a macro, which behaves pretty much like cut and paste. I just did not want to litter the code with `if(XaieLibraryCall() != OK) return not_OK`, and thought a macro was a clean way of capturing this repetitive pattern. I don't think the fact that this code is generated affects how this works.

Reply:
This is a broader discussion related to the specific way the runtime is designed and even the program using it.
For example if at some point MLIR-AIE mix more some CPU and AIE in a single-source way like CUDA, there might be some CPU code calling a completely independent AIE realm and returning a value would not mean anything since it would return from a random function and not `main()` which is the strong assumption made by the runtime today and by your `__mlir_aie_try` implementation.
But forget about it. Outside of this PR.
Otherwise the new code is quite clearer, thanks.
Actually you can even put comments in multi-line macros with `/* */` if you need.